### ISSUE TITLE
Issues with validation in WriteCommentSerializer

### ISSUE THREAD

**Issue by dest81** [danirus/django-comments-xtd#274]
I got errors:
 `AttributeError: 'AnonymousUser' object has no attribute 'email'` [here](https://github.com/danirus/django-comments-xtd/blob/master/django_comments_xtd/api/serializers.py#L60) 
  `AttributeError: 'AnonymousUser' object has no attribute 'get_full_name'` [here](https://github.com/danirus/django-comments-xtd/blob/master/django_comments_xtd/api/serializers.py#L48). 

Check is done in wrong order. First should be checked `user.is_anonymous` and then `user.get_full_name()`. 
 
 Also there are more possible issues with it. Not all user models inherited from AbstractUser and this means that 'email' and 'get_full_name' might not exist. 
 
 I can't test all cases but code below might work:
 
 ```
 from django.contrib.auth import get_user_model

    def validate_name(self, value):
        if not value.strip():
            if self.request.user.is_authenticated:
                if hasattr(self.request.user, "get_full_name"):
                    name = self.request.user.get_full_name()
                else:
                    name = self.request.user.get_username()
                if name:
                    return name

            raise serializers.ValidationError("This field is required")
        return value

    def validate_email(self, value):
        if not value:
            if self.request.user.is_authenticated:
                UserModel = get_user_model()
                email_field = UserModel.get_email_field_name()
                email = getattr(self.request.user, email_field, None)
                if email:
                    return email
            raise serializers.ValidationError("This field is required")
        return value
 ```

**Comment by danirus** [comment]
Hi @dest81, thanks for your report and your solution!

**Comment by danirus** [comment]
The code fix will be release with v2.8.3.
Thanks again for reporting it.

**Comment by dest81** [comment]
@danirus thank you for the quick reaction
